src/pyproject_toml.rs

```diff
@@ -150,6 +149,27 @@ impl PyProjectToml {
         }
         true
     }
+
+    /// Having a pyproject.toml without `build-backend` set to `maturin`
+    /// may result in build errors when build from source distribution
+    ///
+    /// Returns true if the pyproject.toml has `build-backend` set to `maturin`
+    pub fn warn_missing_build_backend(&self) -> bool {
+        if let Some(backend) = self.build_system.build_backend.as_ref() {
+            let maturin = env!("CARGO_PKG_NAME");
+            if backend != maturin {
+                eprintln!(
+                    "⚠️  Warning: `build-backend` in pyproject.toml isn't '{maturin}', \
+                    source distribution may not build properly.",
+                    maturin = maturin
+                );
+                return false;
+            }
+            return true;
+        }
+        eprintln!("⚠️  Warning: missing `build-backend` in pyproject.toml, source distribution may not build properly.");
```

Comment:
These two messages are so similar, maybe it's better to merge them into something like this?

```
eprintln!("⚠️ Warning: `build-backend` in pyproject.toml is not set to `maturin`, source distribution may not build properly.");
```

Also, "source distribution may not build property" seems a bit vague for users to know what will go wrong. I wonder if we can find something more specific to say, maybe "packaging tools such as pip will not use maturin to build this project"?